### PR TITLE
Windows CI: Fixes panic in test-unit for FileUtils

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

This fixes the following panic seen in current Windows CI unit-testing:

```
00:03:53.489 --- FAIL: TestCleanPatternsFolderSplit (0.00s)
00:03:53.489 panic: runtime error: index out of range [recovered]
00:03:53.489    panic: runtime error: index out of range
00:03:53.489 
00:03:53.489 goroutine 34 [running]:
00:03:53.490 testing.tRunner.func1(0xc0820738c0)
00:03:53.490    C:/Go/src/testing/testing.go:450 +0x178
00:03:53.490 github.com/docker/docker/pkg/fileutils.TestCleanPatternsFolderSplit(0xc0820738c0)
00:03:53.490    C:/Go/src/github.com/docker/docker/pkg/fileutils/fileutils_test.go:433 +0x49b
00:03:53.490 testing.tRunner(0xc0820738c0, 0x6dcd20)
00:03:53.490    C:/Go/src/testing/testing.go:456 +0x9f
00:03:53.560 created by testing.RunTests
00:03:53.561    C:/Go/src/testing/testing.go:561 +0x874
00:03:53.561 
00:03:53.561 goroutine 1 [chan receive]:
00:03:53.561 testing.RunTests(0x63a368, 0x6dca80, 0x20, 0x20, 0xc082058700)
00:03:53.561    C:/Go/src/testing/testing.go:562 +0x8b4
00:03:53.561 testing.(*M).Run(0xc082031ee8, 0x6df540)
00:03:53.561    C:/Go/src/testing/testing.go:494 +0x77
00:03:53.561 main.main()
00:03:53.561    github.com/docker/docker/pkg/fileutils/_test/_testmain.go:164 +0x259
00:03:53.561 FAIL   github.com/docker/docker/pkg/fileutils  0.041s
```

@vbatts @swernli I haven't done any level of testing outside of the verifying the unit test doesn't panic when run locally. Would welcome any feedback on if this is a correct change or has other implications. The only caller I can see for `fileutils.CleanPatterns` is `TarWithOptions` in pkg\archive\archive.go, passing `options.ExcludePatterns`. I'm not sure how these `ExcludePatterns` are expected to be formatted.

### CODE REVIEW

**Comment by jhowardmsft** [comment]
@darrenstahlmsft Any chance you could take a look at this as well as it failed win2lin on a build test. I haven't investigated the cause of the failure though yet.

**Comment by icecrime** [comment]
Ping @darrenstahlmsft?

**Comment by darrenstahlmsft** [comment]
Looking now

**Comment by darrenstahlmsft** [comment]
@jhowardmsft 

The problem is in OptimizedMatches just below. This change cleans the patterns to use the os specific path separator, but then attempt to find regex matches with a hardcoded "/"

Changing the hardcoded "/"s in OptimizedMatches to os.PathSeparator fixes the CI (running a full local CI now, I'll update if it fails). I have not tested if this breaks unit tests any more.

File: pkg/fileutils/fileutils.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func CleanPatterns(patterns []string) ([]string, [][]string, bool, error) {
 		if exclusion(pattern) {
 			pattern = pattern[1:]
 		}
-		patternDirs = append(patternDirs, strings.Split(pattern, "/"))
+		patternDirs = append(patternDirs, strings.Split(pattern, string(os.PathSeparator)))
```
Comment (jhowardmsft) [inline pkg/fileutils/fileutils.go]: @cpuguy83 Possibly, but not at all yet convinced.

Comment (jhowardmsft) [inline pkg/fileutils/fileutils.go]: @cpuguy83 No, it's not a filepath.Split - that just gets a single dir and file. This is doing a split of all components.

Comment (cpuguy83) [inline pkg/fileutils/fileutils.go]: Yeah, realized that later :)

**Comment by jhowardmsft** [comment]
userns failure was unrelated. Have kicked it off again

**Comment by jhowardmsft** [comment]
ping @icecrime LGTY now?

**Comment by calavera** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**